pkgs/test_api/lib/src/backend/metadata.dart

```diff
@@ -81,37 +81,37 @@ class Metadata {
     onPlatform.forEach((platform, metadata) {
       if (metadata is Timeout || metadata is Skip) {
         metadata = [metadata];
-      } else if (metadata is! List) {
-        throw ArgumentError('Metadata for platform "$platform" must be a '
-            'Timeout, Skip, or List of those; was "$metadata".');
-      }
-
-      var selector = PlatformSelector.parse(platform);
-
-      Timeout? timeout;
-      dynamic skip;
-      for (var metadatum in metadata) {
-        if (metadatum is Timeout) {
-          if (timeout != null) {
-            throw ArgumentError('Only a single Timeout may be declared for '
-                '"$platform".');
-          }
-
-          timeout = metadatum;
-        } else if (metadatum is Skip) {
-          if (skip != null) {
-            throw ArgumentError('Only a single Skip may be declared for '
-                '"$platform".');
+      } else if (metadata is List) {
```

Comment:
This is a behavior change. When the argument was a `Timeout` it is now effectively ignored I think.

I think this is more readable with an extra helper method.

```dart
  /// Parses a user-provided map into the value for [onPlatform].
  static Map<PlatformSelector, Metadata> _parseOnPlatform(
      Map<String, dynamic>? onPlatform) {
    if (onPlatform == null) return {};

    var result = <PlatformSelector, Metadata>{};
    onPlatform.forEach((platform, metadata) {
      var selector = PlatformSelector.parse(platform);
      if (metadata is Timeout || metadata is Skip) {
        result[selector] = _parsePlatformOptions(platform, [metadata]);
      } else if (metadata is List) {
        result[selector] = _parsePlatformOptions(platform, metadata);
      }
      throw ArgumentError('Metadata for platform "$platform" must be a '
          'Timeout, Skip, or List of those; was "$metadata".');
    });
    return result;
  }

  static Metadata _parsePlatformOptions(
      String platform, List<dynamic> metadata) {
    Timeout? timeout;
    dynamic skip;
    for (var metadatum in metadata) {
      if (metadatum is Timeout) {
        if (timeout != null) {
          throw ArgumentError('Only a single Timeout may be declared for '
              '"$platform".');
        }

        timeout = metadatum;
      } else if (metadatum is Skip) {
        if (skip != null) {
          throw ArgumentError('Only a single Skip may be declared for '
              '"$platform".');
        }

        skip = metadatum.reason ?? true;
      } else {
        throw ArgumentError('Metadata for platform "$platform" must be a '
            'Timeout, Skip, or List of those; was "$metadata".');
      }
    }

    return Metadata.parse(timeout: timeout, skip: skip);
  }
```